src/chunkserver/datastore/file_pool.cpp

```diff
@@ -436,11 +610,13 @@ int FilePool::RecycleFile(const std::string& chunkpath) {
             return -1;
         } else {
             LOG(INFO) << "Recycle " << chunkpath.c_str() << ", success!"
-                      << ", now chunkpool size = " << tmpChunkvec_.size() + 1;
+                      << ", now chunkpool size = "
+                      << currentState_.dirtyChunksLeft + 1;
         }
         std::unique_lock<std::mutex> lk(mtx_);
-        tmpChunkvec_.push_back(newfilenum);
-        ++currentState_.preallocatedChunksLeft;
+        dirtyChunks_.push_back(newfilenum);
```

Comment:
if the chunkserver crashed, will the recycled dirty chunks missed?

Reply:
@SeanHai No, the recycled dirty chunks already in the chunkfile directory, the `dirtyChunks_` vector is only hold the chunk's number in memory.